### PR TITLE
Added a unit test for built-in options docs

### PR DESCRIPTION
**Pull request by TheQwertiest**
As per https://github.com/mesonbuild/meson/pull/4931#discussion_r273429955

### CODE REVIEW

File: run_unittests.py [hunk 1]
```diff
@@ -1008,6 +1008,42 @@ def test_compiler_options_documented(self):
...
+            md = f.read()
+        self.assertIsNotNone(md)
+
+        found_entries = set()
+        sections = list(re.finditer(r"^## (.+)$", md, re.MULTILINE)) + [None]
+
+        for s1, s2 in zip(sections[:], sections[1:]):
+            if s1.group(1) == "Universal options":
+                # Extract the content for this section
+                end = s2.start() if s2 != None else len(md)
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ [E711] comparison to None should be 'if cond is not None:' ([link](https://sider.review/gh/repos/19784232/pulls/5233#issue-3971914))

<sub>posted by Sider</sub>

File: run_unittests.py [hunk 2]
```diff
@@ -1008,6 +1008,42 @@ def test_compiler_options_documented(self):
...
+                content = md[s1.end():end]
+                subsections = list(re.finditer(r"^### (.+)$", content, re.MULTILINE)) + [None]
+
+                for sub1, sub2 in zip(subsections[:], subsections[1:]):
+                    if sub1.group(1) == "Directories" or sub1.group(1) == "Core options":
+                        # Extract the content for this subsection
+                        sub_end = sub2.start() if sub2 != None else len(content)
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ [E711] comparison to None should be 'if cond is not None:' ([link](https://sider.review/gh/repos/19784232/pulls/5233#issue-3971915))

<sub>posted by Sider</sub>

**Comment by nirbheek** [comment]
This branch just needs a rebase now (probably).

**Comment by jpakkane** [comment]
```
FAIL: test_builtin_options_documented (__main__.DataTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_unittests.py", line 1045, in test_builtin_options_documented
    self.assertEqual(found_entries, set(mesonbuild.coredata.builtin_options.keys()))
AssertionError: Items in the second set but not the first:
'wrap_mode'
```

**Comment by TheQwertiest** [comment]
Yep, that's why there was a #5241 =)

**Comment by jpakkane** [comment]
It's merged, so rebasing should make this work.

**Comment by TheQwertiest** [comment]
Not sure if the test failure is related to this PR: https://travis-ci.org/mesonbuild/meson/jobs/518411234#L794

**Comment by TheQwertiest** [comment]
Testing if it's a flaky test

**Comment by TheQwertiest** [comment]
Yup, it was a flaky test =)

**Pull request merged**